app/(main)/league/[leagueId]/entry/all/page.test.tsx

```diff
@@ -65,4 +77,98 @@ describe('Entry Component', () => {
+    mockUseDataStore.mockReturnValueOnce({ user: { id: '123', leagues: [] } });
+    mockGetGameWeek.mockResolvedValueOnce({ week: 1 });
+    mockGetCurrentUserEntries.mockResolvedValueOnce([
+      {
+        id: '66311a210039f0532044',
+        week: 1,
+      },
+    ]);
+    mockGetCurrentLeague.mockResolvedValueOnce({
+      leagueName: 'GiS League',
+      participants: 47,
+      survivors: 47,
+    });
+
+    render(<Entry params={{ leagueId: '66311a210039f0532044' }} />);
+
+    await waitFor(() => {
+      const entryPageHeader = screen.getByTestId('entry-page-header');
+      const entryPageHeaderToLeaguesLink = screen.getByTestId(
+        'entry-page-header-to-leagues-link',
+      );
+      const entryPageHeaderLeagueName = screen.getByTestId(
+        'entry-page-header-league-name',
+      );
+      const entryPageHeaderLeagueSurvivors =
+        screen.getByTestId('LeagueSurvivors');
+      const entryPageHeaderCurrentWeek = screen.getByTestId(
+        'entry-page-header-current-week',
+      );
+
+      expect(entryPageHeader).toBeInTheDocument();
+      expect(entryPageHeaderToLeaguesLink).toBeInTheDocument();
+      expect(entryPageHeaderToLeaguesLink).toHaveAttribute(
+        'href',
+        '/league/all',
+      );
+      expect(entryPageHeaderLeagueName).toBeInTheDocument();
+      expect(entryPageHeaderLeagueName).toHaveTextContent('GiS League');
+      expect(entryPageHeaderLeagueSurvivors).toBeInTheDocument();
+      expect(entryPageHeaderLeagueSurvivors).toHaveTextContent('Survivors');
+      expect(entryPageHeaderCurrentWeek).toBeInTheDocument();
+      expect(entryPageHeaderCurrentWeek).toHaveTextContent('Week 1');
+    });
+  });
+
+  it('should display the header with the league name, survivors, and week number, with a past weeks link', async () => {
+    mockUseDataStore.mockReturnValueOnce({ user: { id: '123', leagues: [] } });
+    mockGetGameWeek.mockResolvedValueOnce({ week: 2 });
+    mockGetCurrentUserEntries.mockResolvedValueOnce([
+      {
+        id: '66311a210039f0532044',
+        week: 2,
+      },
+    ]);
+    mockGetCurrentLeague.mockResolvedValueOnce({
+      leagueName: 'GiS League',
+      participants: 47,
+      survivors: 47,
+    });
+
+    render(<Entry params={{ leagueId: '66311a210039f0532044' }} />);
+
+    await waitFor(() => {
+      const entryPageHeader = screen.getByTestId('entry-page-header');
+      const entryPageHeaderToLeaguesLink = screen.getByTestId(
+        'entry-page-header-to-leagues-link',
+      );
+      const entryPageHeaderLeagueName = screen.getByTestId(
+        'entry-page-header-league-name',
+      );
+      const entryPageHeaderLeagueSurvivors =
+        screen.getByTestId('LeagueSurvivors');
+      const entryPageHeaderCurrentWeek = screen.getByTestId(
+        'entry-page-header-current-week',
+      );
+      const viewPastWeeksLink = screen.getByTestId('past-weeks-link');
```

Comment:
nit: I wonder if there's a way to make this into an object or a class so it's easier/more efficient to grab these for your test?

Reply:
> nit: I wonder if there's a way to make this into an object or a class so it's easier/more efficient to grab these for your test?

@choir27 I'm sure I could, however, I feel the simpler the test is the more readable it is. I'd rather leave this simple than make it complex to save a few lines of code.